### PR TITLE
Namenode ha plugin

### PR DESCRIPTION
**Pull request by agardella**
@captain695 @cburk

### CODE REVIEW

**Comment by pelmers** [comment]
can you remove the .DS_STORE files?

File: rice-namenode/source/main.cc
Comment (nichhk) [inline rice-namenode/source/main.cc:35]: Why are we able to delete this?

File: rice-namenode/source/HAServiceProtocolImpl.cc
Comment (nichhk) [inline rice-namenode/source/HAServiceProtocolImpl.cc]: Destructor is not necessary.

**Review by nichhk — CHANGES_REQUESTED** [review]
We can have everything start in the ACTIVE state. And transitionToActive and transitionToStandby don't need to do anything.

File: vagrant_provision.sh
Comment (pelmers) [inline vagrant_provision.sh:39]: can you remove the lines above this which touch the same files but get overwritten here?
`cat > /home/vagrant/hadoop3/etc/hadoop/core-site.xml <<EOF`
and
`cat > /home/vagrant/hadoop3/etc/hadoop/hdfs-site.xml <<EOF`

File: vagrant_provision.sh
Comment (pelmers) [inline vagrant_provision.sh]: this line is repeated at the end of the file too, delete this one

**Comment by Vegetable26** [comment]
Have we tried the scenario where we have two NameNodes A and B. Kill A, then kill B, then bring back A?

File: rice-namenode/include/HaServiceProtocolImpl.h
Comment (efrederic) [inline rice-namenode/include/HaServiceProtocolImpl.h]: is LEASE_CHECK_TIME ever used?

Comment (nichhk) [reply rice-namenode/include/HaServiceProtocolImpl.h]: Nope leases are gone.

File: rice-namenode/source/main.cc
Comment (agardella) [reply rice-namenode/source/main.cc:35]: Lines 34 and 36 do the same thing

File: vagrant_provision.sh
Comment (agardella) [reply vagrant_provision.sh:39]: I think they're touching different files (the ones in /home/vagrant/hadoop3 as opposed to /home/vagrant/hadoop). It looks like they might be unnecessary anyway, but I'm not totally sure

File: vagrant_provision.sh
Comment (agardella) [reply vagrant_provision.sh]: fixed

File: rice-namenode/include/HaServiceProtocolImpl.h
Comment (agardella) [reply rice-namenode/include/HaServiceProtocolImpl.h]: fixed

File: rice-namenode/source/HAServiceProtocolImpl.cc
Comment (agardella) [reply rice-namenode/source/HAServiceProtocolImpl.cc]: fixed

File: vagrant_provision.sh
Comment (pelmers) [reply vagrant_provision.sh:39]: they are the same file, `hadoop` is just a link to `hadoop3` because of `ln -s hadoop3 hadoop` a few lines above.

**Comment by captain695** [comment]
@Vegetable26 just tried that scenario and it seems to work

**Comment by captain695** [comment]
@nichhk does this look all good to merge?

**Comment by nichhk** [comment]
@captain695 LGTM

**Comment by captain695** [comment]
@nichhk I know you guys have been having things break in develop. Should I wait until that's resolved until I try to merge this in?

**Comment by agardella** [comment]
@nichhk Having all of the namenodes start in active means we can't do a manual failover, and it changes quite a bit of our presentation and demo... it also seems conceptually wrong. What's the reason you wanted that change?

**Pull request merged**
